src/jdk.naming.dns/share/classes/com/sun/jndi/dns/DnsClient.java

```diff
@@ -436,13 +437,17 @@ private byte[] doUdpQuery(Packet pkt, InetAddress server,
                 // Packets may only be sent to or received from this server address
                 InetSocketAddress target = new InetSocketAddress(server, port);
                 udpChannel.connect(target);
-                int pktTimeout = (timeout * (1 << retry));
+                // use 1L below to ensure conversion to long and avoid potential
+                // integer overflow (timeout is an int).
+                // no point in supporting timeout > Integer.MAX_VALUE, clamp if needed
+                long pktTimeout = Math.clamp(timeout * (1L << retry), 0, Integer.MAX_VALUE);
```

Comment:
we may also need to clamp retry to [0,31]. or even less.

Reply:
Agreed - though I don't think it matter much. If the shift overflows to negative `pktTimeout` would become 0 which would be interpreted as wait forever.

Reply:
Santized both `retries` and initial `timeout` prop values in [b651829](https://github.com/openjdk/jdk/pull/20892/commits/b6518291f96244ce12ee1da1d9b740c4bd5b11f6)